sdv/sequential/par.py

```diff
@@ -135,13 +134,53 @@ def get_parameters(self):
         return instantiated_parameters
 
     def add_constraints(self, constraints):
-        """Warn the user that constraints can't be added to the ``PARSynthesizer``."""
-        warnings.warn(
-            'The PARSynthesizer does not yet support constraints. This model will ignore any '
-            'constraints in the metadata.'
-        )
-        self._data_processor._constraints = []
-        self._data_processor._constraints_list = []
+        """Add constraints to the synthesizer.
+
+        For PARSynthesizers only allow a list of constraints that follow these rules:
+
+        1) All constraints must be either for all contextual columns or non-contextual column.
+           No mixing constraints that cover both contextual and non-contextual columns
+        2) No overlapping constraints (there are no constraints that act on the same column)
+        3) No custom constraints
+
+        Args:
+            constraints (list):
+                List of constraints described as dictionaries in the following format:
+                    * ``constraint_class``: Name of the constraint to apply.
+                    * ``constraint_parameters``: A dictionary with the constraint parameters.
+        """
+        context_set = set(self.context_columns)
+        constraint_cols = []
+        for constraint in constraints:
+            constraint_parameters = constraint['constraint_parameters']
+            columns = []
+            for param in constraint_parameters:
+                if 'column_name' in param:
+                    col_names = constraint_parameters[param]
+                    if isinstance(col_names, list):
+                        columns.extend(col_names)
+                    else:
+                        columns.append(col_names)
+            for col in columns:
+                if col in constraint_cols:
+                    raise SynthesizerInputError(
+                        'The PARSynthesizer cannot accommodate multiple constraints '
+                        'that overlap on the same columns.')
+                constraint_cols.append(col)
+
+        all_context = all(col in context_set for col in constraint_cols)
+        no_context = all(col not in context_set for col in constraint_cols)
+
+        if all_context or no_context:
+            super().add_constraints(constraints)
+        else:
+            raise SynthesizerInputError(
```

Comment:
I think we need to add a check here as well that raises an error if someone tries to add a custom constraint dict. Even though they won't be able to load it, if they skip that step and add the dict then this method won't give them any feedback.

Reply:
If we try to run a custom constraint it won't be found due to us not being able to load it without `add_custom_constraint_class` and `load_custom_constraint_classes`. We run into the default `InvalidConstraintsError` for a custom constraint that is not predefined. I think it is suitable and an extra check is not needed. What do you think? @amontanez24 

I also added a test to confirm the error is thrown for adding a custom constraint.

Reply:
makes sense